src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -360,6 +400,81 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
                     return attributeList.WithoutTrivia();
                 }
             }));
+        }
+
+        private static async Task RefactorInitializersAsync(
+            INamedTypeSymbol type,
+            SolutionEditor solutionEditor,
+            ImmutableArray<IPropertySymbol> positionalParameters,
+            CancellationToken cancellationToken)
+        {
+            var symbolReferences = await SymbolFinder
+                .FindReferencesAsync(type, solutionEditor.OriginalSolution, cancellationToken).ConfigureAwait(false);
+            var referenceLocations = symbolReferences.SelectMany(reference => reference.Locations);
+            var projLookup = referenceLocations.ToLookup(refLoc => refLoc.Document.Project.Id);
+            foreach (var (projID, projLocs) in projLookup)
+            {
+                // organize by project first, so we can solve one project at a time
+                var project = solutionEditor.OriginalSolution.GetRequiredProject(projID);
+                if (project.Language != LanguageNames.CSharp)
+                {
+                    // skip non-CSharp projects that reference records
+                    continue;
+                }
+
+                var compilation = await project
+                    .GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var docLookup = projLocs.ToLookup(refLoc => refLoc.Document.Id);
+                foreach (var (docID, docLocs) in docLookup)
+                {
+                    var documentEditor = await solutionEditor
+                        .GetDocumentEditorAsync(docID, cancellationToken).ConfigureAwait(false);
+                    var syntaxTree = await documentEditor.OriginalDocument
+                        .GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var semanticModel = compilation.GetSemanticModel(syntaxTree);
+                    var nodes = docLocs
+                        // we should find the identifier node of an object creation expression
+                        .Select(refLoc => refLoc.Location.FindNode(cancellationToken).Parent)
+                        .OfType<ObjectCreationExpressionSyntax>();
+                    foreach (var node in nodes)
```

Comment:
when doing heavily complex code, i recommend not using names like `node` as it's vague what that refers to.